Notifications/notifications_actions_customization.md

```diff
@@ -12,99 +12,111 @@ This document is a starting point for engaging the community and standards bodie
-- `"critical"`.
-
-These styles are platform-dependent but should be distinct from each other to allow the user to easily identify its options. For example, the `"success"` action button could have a green theme - e.g., have green background color or have a green font over the default background - while the `"critical"` could be red-themed, but it is up to the platform to define the style. If the selected style is not supported by the platform, it should fallback to `"default"`
+We propose the creation of a new property `scenario` in the `NotificationOptions` dictionary of type `NotificationScenario`, that is an `enum` with two values: `"default"` and `"incoming-call"`. The color treatment and ringtone capabilities will only be available in the `"incoming-call"` scenario.
 
 ```javascript
-enum NotificationActionStyle {
+enum NotificationScenario {
   "default",
-  "success",
-  "critical"
+  "incoming-call"
 }
 
-dictionary NotificationAction {
-  required DOMString action;
-  required DOMString title;
-  USVString icon;
-  NotificationActionStyle style = "default";
-};
+dictionary NotificationOptions {
+  NotificationScenario scenario = "default";
+  ...
+}
 ```
 
-Similarly, to support this scenario we also propose the creation of a new NotificationSoundType enum and the inclusion of a `sound` attribute to the Notification interface and a correspondent property to the NotificationOptions dictionary:
+When the scenario is `"incoming-call"`, the notification will always have a default dismiss button. This way, when a calling notification is displayed to the user, we can guarantee that he or she will always be capable to dismiss it. 
 
-```javascript
-enum NotificationSoundType {
-  "default",
-  "ringtone"
-}
+Moreover, notifications created within the `"incoming-call"` scenario should have increased priority over the regular ones. That is, if a `"default"` notification is currently being displayed and another `"incoming-call"` notification gets created, it should break through and be displayed immediately, requiring immediate action from the user.
 
-partial interface Notification {
-  readonly attribute NotificationSoundType sound;
-}
+### Color Treatment
 
-dictionary NotificationOptions {
-  ...
-  NotificationSoundType sound = "default";
+For notifications with `"scenario"` of type `"incoming-call"`, the User Agent (UA) should set the notification's action buttons' color in a way that the style for the default dismiss button is preferably different from the other buttons, which are declared each as a `NotificationAction` element of the `Notfication.actions` array. If the platform does not allow color treatment for notification buttons, the UA should fallback to the `"default"` scenario colors.
+
+We propose that the default dismiss action button should have a red theme - e.g., red background color with some predefined icon on it - while all the other buttons are green-themed, but it is up to the platform to define the style.
```

Comment:
" but it is up to the platform to define the style."

By platform, do we mean the "user agent" here?

Reply:
I actually want to mean the operational system (WIndows, Linux, iOS, Mac, etc). In my conception: platform <=> operational system; user agent <=> browser/application used by the user to access the Web. Is it correct?

Reply:
Yes.